x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportGetShutdownStatusAction.java

```diff
@@ -192,12 +192,6 @@ static ShutdownShardMigrationStatus shardMigrationStatus(
             );
         }
 
-        // The node is in `DiscoveryNodes`, but not `RoutingNodes` - so there are no shards assigned to it. We're done.
-        if (currentState.getRoutingNodes().node(nodeId) == null) {
-            // We don't know about that node
-            return new ShutdownShardMigrationStatus(SingleNodeShutdownMetadata.Status.COMPLETE, 0);
-        }
```

Comment:
This branch prevents us from checking unassigned shards if the node already left the cluster (causing `currentState.getRoutingNodes().node(nodeId)` to be equal to null)